### ISSUE TITLE
Update chapter 4.5.6 on aborting jobs in documentation [BATCH-1788]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-batch#1801]
**[Morten Andersen-Gott](https://jira.spring.io/secure/ViewProfile.jspa?name=magott)** opened **[BATCH-1788](https://jira.spring.io/browse/BATCH-1788?redirect=false)** and commented

Chapter 4.5.6 [1] says that a killed job (kill -9) can be set to be either aborted or failed. Where failed will allow restart by the framework, while aborted will not.

First of, there is no BatchStatus.ABORTED, there is however a BatchStatus.ABANDONED, I guess this is what the documentation should say?

Second, the BatchStatus.ABANDONED can be restarted by the CommandLineJobRunner by providing the -restart flag. The CLJR will restart any job where status greater than STOPPING. STOPPED, FAILED, ABANDONED and UNKNOWN are greater than STOPPING and will thus be restartable, provided that the job is restartable=true.

If you try to restart an abandoned batch by supplying the identical jobparameters and job name for an abandoned job, the SimpleJobRepository will throw an exception saying "A job instance already exists and is complete for parameters..."

In other words, there is a discrepancy in how what the CommandLineJobRunner and the SimpleJobRepository allows to be restarted.

[1] http://static.springsource.org/spring-batch/reference/html-single/index.html#d0e2450

---

**Referenced from:** commits https://github.com/spring-projects/spring-batch/commit/a06f4ad38958f844226ab4bd2205ca1ba841800f, https://github.com/spring-projects/spring-batch/commit/9efb3c0dac9ef59ac9fb2dcafd9f0e8ecc4a6901

**Backported to:** [2.2.0 - Sprint 14](https://github.com/spring-projects/spring-batch/milestone/57?closed=1)

1 votes, 2 watchers
